paperqa/llms.py

```diff
@@ -875,12 +910,56 @@ def clear(self) -> None:
         super().clear()
         self.texts = []
         self._embeddings_matrix = None
+        self._texts_filter = None
 
     def add_texts_and_embeddings(self, texts: Iterable[Embeddable]) -> None:
         super().add_texts_and_embeddings(texts)
         self.texts.extend(texts)
         self._embeddings_matrix = np.array([t.embedding for t in self.texts])
 
+    async def partitioned_similarity_search(
+        self,
+        query: str,
+        k: int,
+        embedding_model: EmbeddingModel,
+        partitioning_fn: Callable[[Doc], int],
+    ) -> tuple[Sequence[Embeddable], list[float]]:
+        scores = []
+        texts = []
```

Comment:
Can you type annotate these? It's nice to know if:
- `scores` is a `float`, `int`, or `str`
- `texts` are a `str` or `Text`